python/cuml/dask/decomposition/tsvd.py

```diff
@@ -0,0 +1,412 @@
+        If it is None, a new one is created just for this class
+    n_components : int (default = 1)
+        The number of top K singular vectors / values you want.
+        Must be <= number(columns).
+    svd_solver : 'full'
+        Only Full algorithm is supported since it's significantly faster on GPU
+        then the other solvers including randomized SVD.
+    verbose : bool
+        Whether to print debug spews
+    
+    Attributes
+    ----------
+    components_ : array
+        The top K components (VT.T[:,:n_components]) in U, S, VT = svd(X)
+    explained_variance_ : array
+        How much each component explains the variance in the data given by S**2
+    explained_variance_ratio_ : array
+        How much in % the variance is explained given by S**2/sum(S**2)
+    singular_values_ : array
+        The top K singular values. Remember all singular values >= 0
+    """
+
+    def __init__(self, client=None, **kwargs):
+        """
+        Constructor for distributed TruncatedSVD model
+        """
+        self.client = default_client() if client is None else client
+        self.kwargs = kwargs
+
+        # define attributes to make sure they
+        # are available even on untrained object
+        self.local_model = None
+        self.components_ = None
+        self.explained_variance_ = None
+        self.explained_variance_ratio_ = None
+        self.singular_values_ = None
+
+    @staticmethod
+    def _func_create_model(sessionId, dfs, **kwargs):
+        try:
+            from cuml.decomposition.tsvd_mg import TSVDMG as cumlTSVD
+        except ImportError:
+            raise Exception("cuML has not been built with multiGPU support "
+                            "enabled. Build with the --multigpu flag to"
+                            " enable multiGPU support.")
+
+        handle = worker_state(sessionId)["handle"]
+        return cumlTSVD(handle=handle, **kwargs), dfs
+
+    @staticmethod
+    def _func_fit(f, M, N, partsToRanks, rank, transform):
+        m, dfs = f
+        return m.fit(dfs, M, N, partsToRanks, rank, transform)
+
+    @staticmethod
+    def _func_transform(f, df, M, N, partsToRanks, rank):
+        m, dfs = f
+        return m.transform(df, M, N, partsToRanks, rank)
+
+    @staticmethod
+    def _func_inverse_transform(f, df, M, N, partsToRanks, rank):
+        m, dfs = f
+        return m.inverse_transform(df, M, N, partsToRanks, rank)
+
+    @staticmethod
+    def _func_get_first(f):
+        return f[0]
+
+    @staticmethod
+    def _func_get_idx(f, idx):
+        return f[idx]
+
+    @staticmethod
+    def _func_xform(model, df):
+        return model.transform(df)
+
+    @staticmethod
+    def _func_get_size(df):
+        return df.shape[0]
+
+    def fit(self, X, _transform=False):
+        """
+        Fit the model with X.
+
+        Parameters
+        ----------
+        X : dask cuDF input
+
+        """
+        gpu_futures = self.client.sync(extract_ddf_partitions, X, agg=False)
```

Comment:
@cjnolet this PR still uses the `agg` parameter that PR #1282 removes. Depending on merge timing, either it'll need to be corrected here or in that PR. Tagging @cjnolet

Reply:
Tried to remove agg param but getting an error. Will update this part after this is merged.